src/kudu/consensus/raft_consensus.cc

```diff
@@ -3586,6 +3586,7 @@ void RaftConsensus::HandleProxyRequest(const ConsensusRequestPB* request,
 
     vector<ReplicateRefPtr> messages;
     do {
+      messages.clear();
 
       // We assume and enforce that a single request is composed of a range of ops.
       int64_t first_op_index = -1;
```

Comment:
line 3606 in the commit: In what case would the message ids in the request not be contiguous?

Reply:
> line 3606 in the commit: In what case would the message ids in the request not be contiguous?

Either in the case that we were incorrectly caching them at this (the RaftConsensus / proxying) layer or possibly a corruption in the Log implementation or even the on-disk log itself.